qemu/tests/rh_kselftests_vm.py

```diff
@@ -0,0 +1,41 @@
+from virttest import error_context
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    rh_kselftests_vm test
+    1) Download the current kernel selftests RPM
+    2) Install the RPM
+    3) Execute the kernel selftests
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    kernel_path = params.get("kernel_path", "/tmp/kernel")
+    tests_execution_cmd = params.get("tests_execution_cmd")
+
+    session.cmd("mkdir -p %s" % kernel_path)
+    kernel_version = session.cmd_output("uname -r").strip().split("+")[0]
+    error_context.base_context("The kernel version: %s" % kernel_version, test.log.info)
+
+    error_context.context("Download the kernel selftests RPM", test.log.debug)
+    session.cmd("cd %s" % kernel_path)
+    session.cmd(
+        "brew download-build --rpm kernel-selftests-internal-%s.rpm" % kernel_version,
+        180,
+    )
+
+    error_context.context("Install the RPM", test.log.debug)
+    session.cmd("dnf install -y ./kernel-*")
```

Comment:
Not a comment, I am OK with the code, just a reminder
1. It may be better to do a check before the installation, of course, dnf will succeed even if the package is already installed, so we don't hit error here
2. It may be better to use a function(e.g. mm_test) to encapsulate your mm test steps. In future, when we add a new test (a new variant in cfg), we need to do this, but currently, it's OK for only mm is covered.

Reply:
Totally agree with @zhencliu for the situation 2 above.

Reply:
@zhencliu thanks for the heads up, with the second statement you mean using a function instead of executing the value of `test_execution_cmd` right? I would agree with this if the next variant to be added needs a different approach to handle the selftests execution, otherwise just overwritting that cfg value should be enough

Reply:
@mcasquer  qemu_guest_agent can be a reference, it use the `gagent_check_type` to distinguish, and corresponding function for the test.

Reply:
@zhencliu  I think about this again,  and i wonder if all the tests is running a script command.  
If yes,  I am ok with the current solution now,   `tests_execution_cmd` is enough i think.
If no,  yes, individual function for each test is better.

Reply:
hi @mcasquer and @yanan-fu , as I said, I am OK with the current code, just a reminder. 

Consider this situation, we have to do some pre-work before running the tests_execution_cmd, we still need to use at least a function, actually even for this test, we set setup_hugepages=yes, ie. VT helps to do the preparation. If VT doesn't support it, we have to do the preparation ourselves. 

Just some thoughts, again, I am OK with the code, thanks for your response.

Reply:
Agree,  including pre setup steps in test case level,   let's use individual function ~